src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunReader.cs

```diff
@@ -756,7 +756,7 @@ private void ParseMethodDefEntrypoints(Action<ReadyToRunSection, IAssemblyMetada
                 {
                     if (ReadyToRunAssemblyHeaders[assemblyIndex].Sections.TryGetValue(ReadyToRunSectionType.MethodDefEntryPoints, out methodEntryPointSection))
                     {
-                        methodDefSectionReader(methodEntryPointSection, OpenReferenceAssembly(assemblyIndex + 1));
+                        methodDefSectionReader(methodEntryPointSection, OpenReferenceAssembly(assemblyIndex + 2));
```

Comment:
You need to make this change under a check for the R2R version. The latest version of R2RDump needs to be able to read old files.

Reply:
Thanks David for pointing that out, fixed in 2nd commit.